sdk/src/account.rs

```diff
@@ -21,6 +22,190 @@ pub struct Account {
     pub rent_epoch: Epoch,
 }
 
+//pub type AccountNoData = Account;
+
+#[derive(PartialEq, Eq, Clone, Default, AbiExample)]
+pub struct AccountNoData {
+    /// lamports in the account
+    pub lamports: u64,
+    /// data held in this account
+    pub data: Arc<Vec<u8>>,
```

Comment:
This is the root change in this pr. data becomes an Arc.

Reply:
@Lichtso fyi. We can discuss this Freitag.

Reply:
Is there a reason the whole `Account` doesn't move under `Arc`?

Reply:
We imagine the eventual state may be AccountNoData = Arc<AccountInner>
where AccountInner has an Arc<Data>

The thinking is the data is large and we want to avoid copying it. The account itself is moderate, but fixed size and we copy it a lot. So, if a program changes lamports, the account data changes, but we don't want to copy the data. I was trying to move things in easier to understand stages.
Moving the whole account under and Arc was a next thing to do. I have not built it out and made it work yet to determine the performance differences or the difficulty in getting it to pass the CI.

Reply:
But another next state is where AccountNoData is an enum which either owns the data or it references data from a slice where the slice was the serialized modified accounts sent to a program. I talked with Alex and Carl about this today.

Reply:
Feel free to provide feedback and discuss more. You (Jack) were on my list to discuss with but I haven't been able to connect yet.

Reply:
I just tried this:
`
pub type AccountNoData = Arc<AccountNoDataInner>;
    
#[derive(PartialEq, Eq, Clone, Default, AbiExample)]
pub struct AccountNoDataInner {
    /// lamports in the account
    pub lamports: u64,
    /// data held in this account
    pub data: Arc<Vec<u8>>,
    /// the program that owns this account. If executable, the program that loads this account.
    pub owner: Pubkey,
    /// this account's data contains a loaded program (and is now read-only)
    pub executable: bool,
    /// the epoch at which this account will next owe rent
    pub rent_epoch: Epoch,
}

`
It didn't seem to quite work easily. I have more Rust to learn still. I was imagining we'd want to move the members of AccountNoData to private and use the AnAccount trait to access them. That allows us later to abstract away that the data is stored in a slice from a program's execution. Moving fields to private was another follow on pr I was imagining shortly after these 2 went in.

Reply:
Attempt 2 at Arc would be making AccountNoData a struct with a single private member that was an Arc. Once again, I have not measured this. I don't know how to quantify how much the additional arc affects performance. I know the data Arc helps.

Reply:
The account is a unit that reflects a whole state, are there cases where we want to specifically copy parts (say lamports) but share others (data)?  If, not it seems more clear to pass a shared copy around and only do a deep copy if necessary.  I'm available Monday if you'd like to talk it out